include/nbl/builtin/hlsl/scene/keyframe.hlsl

```diff
@@ -0,0 +1,54 @@
+
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SCENE_KEYFRAME_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SCENE_KEYFRAME_INCLUDED_
+
+
+
+#include <nbl/builtin/hlsl/math/quaternions.hlsl>
+
+
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace scene
+{
+	using namespace math;
+
+
+	struct Keyframe_t
+	{
+		uint2 data[3];
+
+		float3 getScale();
+		quaternion_t getRotation();
+		float3 getTranslation();

```

Comment:
you need to put the function bodies back here

Reply:
I thought we are writing implementation file separately

Reply:
did you even read https://github.com/Devsh-Graphics-Programming/Nabla/pull/428#issue-1435947656 ?